src/app/features/tasks/store/task.reducer.ts

```diff
@@ -113,6 +113,24 @@ export const taskReducer = createReducer<TaskState>(
     });
   }),
 
+  on(toggleHideFromMenu, (state, { projectId, allTaskIds, isHiddenFromMenu }) => {
```

Comment:
I'd rather avoid updating the task model itself for this, since it might complicate things further down the road, adds additional data size and logically the hidden state belongs more to the project model, than the task model.